lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -81,7 +81,13 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
         }
 
         ApacheHttpClient.Builder httpClient = ApacheHttpClient.builder()
-                .maxConnections(config.getMaxConnections());
+                .maxConnections(config.getMaxConnections())
+                .tcpKeepAlive(config.getTcpKeepAlive());
+
+        config.getConnectionTtl().ifPresent(connectionTtl -> httpClient.connectionTimeToLive(connectionTtl.toJavaTime()));
```

Comment:
I think we could simplify these like

```java
config.getConnectionTtl().map(Duration::toJavaTime).ifPresent(httpClient::connectionTimeToLive);

Reply:
We could, but that'll be a lot of method handle instantiation and extra line noise compared to what's there which doesn't seem that much worse readability wise to me.